src/rpc/mempool.cpp

```diff
@@ -669,6 +670,136 @@ static RPCHelpMan savemempool()
+                {RPCResult::Type::ARR, "replaced-transactions", /*optional=*/ true, "List of txids of replaced transactions",
+                {
+                    {RPCResult::Type::STR_HEX, "", "The transaction id"},
+                }},
+            },
+        },
+        RPCExamples{
+            HelpExampleCli("testmempoolaccept", "[rawtx1, rawtx2]") +
+            HelpExampleCli("submitrawtransaction", "[rawtx1, rawtx2]")
+        },
+        [&](const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+    {
+    if (!Params().IsMockableChain()) {
+        throw std::runtime_error("submitpackage is for regression testing (-regtest mode) only");
+    }
+    RPCTypeCheck(request.params, {
+        UniValue::VARR,
+    });
+    const UniValue raw_transactions = request.params[0].get_array();
+    if (raw_transactions.size() < 1 || raw_transactions.size() > MAX_PACKAGE_COUNT) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER,
+                           "Array must contain between 1 and " + ToString(MAX_PACKAGE_COUNT) + " transactions.");
+    }
+
+    std::vector<CTransactionRef> txns;
+    txns.reserve(raw_transactions.size());
+    for (const auto& rawtx : raw_transactions.getValues()) {
+        CMutableTransaction mtx;
+        if (!DecodeHexTx(mtx, rawtx.get_str())) {
+            throw JSONRPCError(RPC_DESERIALIZATION_ERROR,
+                               "TX decode failed: " + rawtx.get_str() + " Make sure the tx has at least one input.");
+        }
+        txns.emplace_back(MakeTransactionRef(std::move(mtx)));
+    }
+
+    NodeContext& node = EnsureAnyNodeContext(request.context);
+    CTxMemPool& mempool = EnsureMemPool(node);
+    CChainState& chainstate = EnsureChainman(node).ActiveChainstate();
+    const auto package_result = WITH_LOCK(::cs_main, return ProcessNewPackage(chainstate, mempool, txns, /* test_accept */ false));
+
+    // First catch any errors.
+    switch(package_result.m_state.GetResult()) {
+        case PackageValidationResult::PCKG_RESULT_UNSET: break;
+        case PackageValidationResult::PCKG_POLICY:
+        {
+            throw JSONRPCTransactionError(TransactionError::MEMPOOL_ERROR,
+                package_result.m_state.GetRejectReason());
+        }
+        case PackageValidationResult::PCKG_MEMPOOL_ERROR:
+        {
+            throw JSONRPCTransactionError(TransactionError::INVALID_PACKAGE,
+                package_result.m_state.GetRejectReason());
+        }
+        case PackageValidationResult::PCKG_TX:
+        {
+            for (const auto& tx : txns) {
+                auto it = package_result.m_tx_results.find(tx->GetWitnessHash());
+                if (it != package_result.m_tx_results.end() && it->second.m_state.IsInvalid()) {
+                    throw JSONRPCTransactionError(TransactionError::MEMPOOL_REJECTED,
+                        strprintf("%s failed: %s", tx->GetHash().ToString(), it->second.m_state.GetRejectReason()));
+                }
+            }
```

Comment:
Should we either `Assume(false)` here or raise a `JSONRPCTransactionError(TransactionError::MEMPOOL_ERROR)`?  There shouldn't be a `PackageValidationResult::PCKG_TX` without one of the transactions having an invalid status.

Reply:
Good point, added `Assume(false)` since it would be a code mistake